server/master/src/main/java/org/apache/accumulo/master/tableOps/Utils.java

```diff
@@ -59,21 +57,16 @@ static void checkTableDoesNotExist(Instance instance, String tableName, Table.ID
       throw new AcceptableThriftTableOperationException(null, tableName, operation, TableOperationExceptionType.EXISTS, null);
   }
 
-  static <T extends AbstractId> T getNextTableId(String tableName, Instance instance, Class<T> idClassType) throws AcceptableThriftTableOperationException {
+  static String getNextTableId(String tableName, Instance instance) throws AcceptableThriftTableOperationException {
```

Comment:
It's not different, but it prevents passing a String around (which I believe was the original reason you added the reflection code), but without reflection (lambdas are faster). It's up to you. I like the elegance of the functional stuff, but I don't think it matters much, as long as we're avoiding reflection.